drivers/include/ph_oem.h

```diff
@@ -0,0 +1,376 @@
+ */
+typedef enum {
+    PH_OEM_OK                   = 0,    /**< Everything was fine */
+    PH_OEM_NODEV                = -1,   /**< No device found on the bus */
+    PH_OEM_READ_ERR             = -2,   /**< Reading to device failed*/
+    PH_OEM_WRITE_ERR            = -3,   /**< Writing to device failed */
+    PH_OEM_READ_WRITE_ERR       = -4,   /**< Reading or writing to device failed */
+    PH_OEM_NOT_PH               = -5,   /**< Not an Atlas Scientific pH OEM device */
+    PH_OEM_INTERRUPT_GPIO_UNDEF = -6,   /**< Interrupt pin is @ref GPIO_UNDEF */
+    PH_OEM_ENABLE_GPIO_UNDEF    = -7,   /**< Enable pin is @ref GPIO_UNDEF */
+    PH_OEM_GPIO_INIT_ERR        = -8    /**< Error while initializing GPIO PIN */
+} ph_oem_named_returns_t;
+
+/**
+ * @brief   LED state values
+ */
+typedef enum {
+    PH_OEM_LED_ON   = 0x01, /**< LED on state */
+    PH_OEM_LED_OFF  = 0x00, /**< LED off state */
+} ph_oem_led_state_t;
+
+/**
+ * @brief   Device state values
+ */
+typedef enum {
+    PH_OEM_TAKE_READINGS    = 0x01, /**< Device active state */
+    PH_OEM_STOP_READINGS    = 0x00, /**< Device hibernate state */
+} ph_oem_device_state_t;
+/**
+ * @brief   Interrupt pin option values
+ */
+typedef enum {
+    PH_OEM_IRQ_DISABLED = 0x00, /**< Interrupt pin option disabled */
+    PH_OEM_IRQ_RISING   = 0x02, /**< Pin high on new reading (manually reset) */
+    PH_OEM_IRQ_FALLING  = 0x04, /**< Pin low on new reading (manually reset) */
+    PH_OEM_IRQ_BOTH     = 0x08, /**< Invert state on new reading (automatically reset) */
+} ph_oem_irq_option_t;
+
+/**
+ * @brief   Calibration option values
+ */
+typedef enum {
+    PH_OEM_CALIBRATE_LOW_POINT  = 0x02,     /**< Low point calibration option */
+    PH_OEM_CALIBRATE_MID_POINT  = 0x03,     /**< Mid point calibration option */
+    PH_OEM_CALIBRATE_HIGH_POINT = 0x04,     /**< High point calibration option */
+} ph_oem_calibration_option_t;
+
+/**
+ * @brief   pH OEM sensor params
+ */
+typedef struct ph_oem_params {
+    i2c_t i2c;              /**< I2C device the sensor is connected to */
+    uint8_t addr;           /**< the slave address of the sensor on the I2C bus */
+    gpio_t interrupt_pin;   /**< interrupt pin (@ref GPIO_UNDEF if not defined) */
+    gpio_t enable_pin;      /**< enable pin (@ref GPIO_UNDEF if not defined) */
+} ph_oem_params_t;
+
+/**
+ * @brief   pH OEM interrupt pin callback
+ */
+typedef void (*ph_oem_interrupt_pin_cb_t)(void *);
+
+/**
+ * @brief   pH OEM device descriptor
+ */
+typedef struct ph_oem {
+    ph_oem_params_t params;         /**< device driver configuration */
+    ph_oem_interrupt_pin_cb_t cb;   /**< interrupt pin callback */
+    void *arg;                      /**< interrupt pin callback param */
+} ph_oem_t;
+
+/**
+ * @brief   Initialize a pH OEM sensor
+ *
+ * @param[in,out]   dev      device descriptor
+ * @param[in]       params   device configuration
+ *
+ * @return zero on successful initialization, non zero on error
```

Comment:
Here and bellow please document with the defined return codes

Reply:
added the defined return codes to the documentation